npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/vulnerability_auditor.rb

```diff
@@ -86,15 +86,21 @@ def audit(dependency:, security_advisories:)
 
         attr_reader :dependency_files, :credentials
 
-        def valid_audit_result?(audit_result, security_advisories)
-          # we only need to check results that indicate a fix is available
-          return true unless audit_result["fix_available"]
+        def viable_audit_result?(audit_result, security_advisories)
+          validation_result = validate_audit_result(audit_result, security_advisories)
+          return true if validation_result == :valid
 
-          return false if vulnerable_dependency_removed?(audit_result)
-          return false if dependency_still_vulnerable?(audit_result, security_advisories)
-          return false if downgrades_dependencies?(audit_result)
+          Dependabot.logger.info("VulnerabilityAuditor: audit result not viable: #{validation_result}")
+          false
+        end
+
+        def validate_audit_result(audit_result, security_advisories)
+          return :fix_unavailable unless audit_result["fix_available"]
+          return :vulnerable_dependency_removed if vulnerable_dependency_removed?(audit_result)
+          return :dependency_still_vulnerable if dependency_still_vulnerable?(audit_result, security_advisories)
+          return :downgrades_dependencies if downgrades_dependencies?(audit_result)
```

Comment:
I really like these reasons, especially the last three! I these will be really helpful for users and maintainers 😄 

🤔 Is `:fix_unavailable` as granular as we can get? It's a big improvement, but I wonder if we could send a signal like `:currently_unpatched` or `:not_fixed`.

Reply:
I think there may be a few edge cases we'd need to detect and expose in the `audit_result` to give a clearer reason when there isn't a fix available. We're planning to look at improving the error message in that case so can dig into that more then.